frontend/stores/orders-store.ts

```diff
@@ -0,0 +1,36 @@
+import { create } from 'zustand'
+import { vegaOrder } from '@vegaprotocol/rest-clients/dist/trading-data'
+import { SendMessage } from '../contexts/json-rpc/json-rpc-provider.tsx'
+import { RpcMethods } from '../lib/client-rpc-methods.ts'
+
+export type OrdersStore = {
+  loading: boolean
+  getOrderById: (id: string, request: SendMessage) => Promise<{ order: vegaOrder; lastUpdated: number }>
+}
+
+export const useOrdersStore = create<OrdersStore>((set) => ({
+  loading: false,
+  async getOrderById(id: string, request: SendMessage) {
+    try {
+      set({ loading: true })
+      const response = await request(RpcMethods.Fetch, { path: `api/v2/order/${id}` })
+      const order = response.order
+
+      if (!order) {
+        throw new Error(`Order with id ${id} not found`)
+      }
+
+      return {
+        order,
+        lastUpdated: Date.now()
+      }
```

Comment:
Rather than returning let's set this on the store as that is the pattern zustand advocates for and in case we want to split components later this removed the need to prop drill.

Reply:
I thought about setting it on the store, but the info is really only pertinent to the specific order that's returned, there's no added benefit for there to be access to the value anywhere else, it just creates more code requiring coverage, ensuring old updatedAt data is ignored etc.